src/hooks/useTokenBalances.ts

```diff
@@ -59,8 +59,15 @@ async function _getBalances(walletInfo: WalletInfo): Promise<TokenBalanceDetails
   const contractAddress = depositApi.getContractAddress()
   const tokens = tokenListApi.getTokens(networkId)
 
-  const balancePromises = tokens.map(async token => fetchBalancesForToken(token, userAddress, contractAddress))
-  return Promise.all(balancePromises)
+  const balancePromises: Promise<TokenBalanceDetails | null>[] = tokens.map(async token =>
+    fetchBalancesForToken(token, userAddress, contractAddress).catch(e => {
+      log('Error for', token, userAddress, contractAddress)
+      log(e)
+      return null
+    }),
+  )
+  const balances = await Promise.all(balancePromises)
+  return balances.filter(Boolean)
```

Comment:
what does Boolean do here? just check if it's a boolean response and filter those out?

Reply:
filters out falsy values
Boolean is equivalent to `val => !!val`